DRAFT Add pre-commit (#40)
ewels/rich-click: README.md

Please create a new [issue](https://github.com/ewels/rich-click/issues)
or better still, dive right in with a pull-request.

> ### Local setup

Reviewer: It would be great to have a bit of prose text about the different code style linters - at least listing what we're using and what they do.

Author: Good idea. Added something.

Please create a new [issue](https://github.com/ewels/rich-click/issues)
or better still, dive right in with a pull-request.

### Local setup

1. Create a new venv with a python3.6+ interpreter using `python3 -m venv venv`
2. Activate the venv with `source venv/bin/activate`
3. Install our the package as an editable including all dev dependencies with `pip3 install -e ."[dev]"`
4. Install pre-commit with `pre-commit install`

#### Pre-commit

Our pre-commit hooks contain the following hooks:

> - prettier: formats our yaml files nicely.
> - no relative imports: prevents you from using relative imports.
> - iSort: will automatically sort the imports alphabetically.
> - black: will automatically format your code to be according to standardized python format.
> - flake8: will do linting checks to make sure all your code is correctly styled and used.
> - mypy: static type checker which verifies you are not using objects incorrectly.

Reviewer: ```suggestion
- [Prettier](https://prettier.io/): formats our markdown and yaml files nicely.
- no relative imports: prevents you from using relative imports.
- [iSort](https://pycqa.github.io/isort/): will automatically sort the imports alphabetically.
- [black](https://black.readthedocs.io/): will automatically format your code to be according to standardized python format.
- [flake8](https://flake8.pycqa.org/): will do linting checks to make sure all your code is correctly styled and used.
- [mypy](http://mypy-lang.org/): static type checker which verifies you are not using objects incorrectly.
```

As mentioned, some of these tools automatically fix your code while other only highlight potential issues.

> Therefore, sometimes it enough to try to commit a second time, and it will pass, while others will require

Reviewer: ```suggestion
Sometimes it will be enough to try to commit a second time and it will pass, while other times it may require
```

manual changes to your code.

> 

Reviewer: ```suggestion

In rare cases it may be difficult or undesirable to change to code to pass the linting rules.
If this happens, it's ok to add a flake8 `# noqa` or mypy `# type: ignore` comment to skip that line.
For details of how to do this, please see the [flake8 docs](https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors)
and [mypy docs](https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker).
```

> ## Credits

Reviewer: Too far off the bottom of the diff to add a suggestion in the PR, but you should add yourself!

Author: > Oops, sorry about the ❌ It's amazingly difficult to write valid markdown in the GitHub PR plain text editor 😆
> 
> Note that for next time you can batch suggestions together if you want instead of doing one commit per comment. This option is only available on the _Files changed_ PR tab though, not _Overview_.

Haha let me just do it on my macbook then :p
